pkgs/development/python-modules/hypercorn/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitLab
+, isPy27, isPy35, isPy36
+, pythonOlder
+, typing-extensions
+, wsproto
+, toml
+, h2
+, priority
+, tox
+}:
+
+buildPythonPackage rec {
+  pname = "Hypercorn";
+  version = "0.11.2";
+  disabled = isPy27 || isPy35 || isPy36;
+
+  src = fetchFromGitLab {
+    owner = "pgjones";
+    repo = pname;
+    rev = version;
+    sha256 = "0v80v6l2xqac5mgrmh2im7y23wpvz4yc2v4h9ryhvl88c2jk9mvh";
+  };
+
+  propagatedBuildInputs = [ wsproto toml h2 priority ]
+  ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
+
+  checkInputs = [ tox ];
+
+  checkPhase = "tox -e py37,py38,py39 -s";
```

Comment:
shouldn't need to use tox directly. As nix already parameterizes the interpreter version.

Reply:
please run the code which they tell tox to run.

Reply:
They tell to just use `tox`, but if I do it like `checkPhase = "tox"` it fails to test and build because of this error:
```
ERROR:   docs: could not install deps [git+https://github.com/pandas-dev/pydata-sphinx-theme.git@master, sphinx, trio]; v = InvocationError('/private/tmp/nix-build-python3.9-Hypercorn-0.11.2.drv-0/source/.tox/docs/bin/python -m pip install git+https://github.com/pandas-dev/pydata-sphinx-theme.git@master sphinx trio', 1)
  format: commands succeeded
ERROR:   mypy: commands failed
ERROR:  py37: InterpreterNotFound: python3.7
ERROR:  py38: InterpreterNotFound: python3.8
  py39: commands succeeded
  pep8: commands succeeded
  setuppy: commands succeeded
ERROR:   manifest: commands failed

builder for '/nix/store/lfcgi79c2gsahnyx5bd9vxhblphwvmpp-python3.9-Hypercorn-0.11.2.drv' failed with exit code 1
```
I build it with command `nix-build -I $NIXPKGS -A python39Packages.hypercorn`.

Reply:
I meant, `tox` will just follow what `tox.ini` has in it. So you should look at https://gitlab.com/pgjones/hypercorn/-/blob/master/tox.ini to see what it's doing

Reply:
You could probably use `pytestCheckHook` and `hypothesis`

Reply:
Why not just use tox, as hypercorn uses and it works?

Reply:
> Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.
> 
> `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)
> 
> should have two commits:
> 
> ```
> maintainers: add dgliwka
> pythonPackages.hypercorn: init at 0.11.2
> ```

Thank you, done it. But it's on my fork: https://github.com/dgliwka/nixpkgs/commits/master .
I do not know how to push it here.

Reply:
Since I could not manage to push code to this pr I opened new one, that complies with CONTRIBUTING.md. Thank you for your help.
https://github.com/NixOS/nixpkgs/pull/123905